### PR TITLE
Companies model DBand Validations

### PR DESCRIPTION
**Pull request by DenisRem**
**CHECK LIST**
- [x] Create model
- [x] Create enum field
- [x] code coverage >90%
- [x] Validation
- [x] seed.rb file is updated if needed

### CODE REVIEW

File: config/database.yml [hunk 1]
```diff
@@ -23,7 +23,7 @@ default: &default
```
Comment (mmarusyk) [inline config/database.yml:23]: Don't use your information about your database in this file or add in .gitconfig

File: app/models/company.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
+class Company < ApplicationRecord
+  enum company_type: { water_supply: 0, heating: 1, rent_payment: 2, accommodation_payment: 3,
+                       gas: 4, electricity: 5, garbage_removal: 6, internet: 7, intercom: 8,
+                       tv: 9, elevator: 10, other: 11 }
+  before_save { email.downcase! }
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/.freeze
+  VALID_PHONE = /\A\d{10}\z/.freeze
+  VALID_FAX = /\A\d{10}\z/.freeze
```
Comment (mmarusyk) [inline app/models/company.rb:8]: Move contstants to the top.

File: app/views/companies/new.html.erb [hunk 3]
```diff
@@ -0,0 +1,2 @@
+<h1>Companies#new</h1>
+<p>Find me in app/views/companies/new.html.erb</p>
```
Comment (mmarusyk) [inline app/views/companies/new.html.erb:2]: We use slim template

File: db/seeds.rb [hunk 4]
```diff
@@ -5,3 +5,6 @@
 #
 #   movies = Movie.create([{ name: 'Star Wars' }, { name: 'Lord of the Rings' }])
 #   Character.create(name: 'Luke', movie: movies.first)
+require 'factory_bot'
+
+FactoryBot.create_list(:company, 50)
```
Comment (mmarusyk) [inline db/seeds.rb:10]: add new line at end of file

File: spec/models/company_spec.rb [hunk 5]
```diff
@@ -0,0 +1,51 @@
+require 'rails_helper'
+
+RSpec.describe Company, type: :model do
+  let!(:company) { FactoryBot.build(:company, name: "IFBut", address_id: 1, company_type: 7, account_id: 7, phone: 2476316898, email: "ifbut@gmail.com", website: "ifbut.com", fax: 9762578517) }
```
Comment (pereslop) [inline spec/models/company_spec.rb:4]: do you need such big list of company properties if you don't use them?

**Review by pereslop — CHANGES_REQUESTED** [review]

File: config/database.yml (hunk 1 above)
Comment (chu4) [reply config/database.yml:23]: thats a common config for every one dont edit it without specific need

File: db/migrate/20200930175155_create_companies.rb [hunk 6]
```diff
@@ -0,0 +1,17 @@
+class CreateCompanies < ActiveRecord::Migration[6.0]
+  def change
+    create_table :companies do |t|
+      t.string :name
+      t.integer :address_id
```
Comment (chu4) [inline db/migrate/20200930175155_create_companies.rb:5]: you dont have this field in you task remove

File: db/migrate/20200930175155_create_companies.rb [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+      t.integer :company_type
+      t.integer :account_id
```
Comment (chu4) [inline db/migrate/20200930175155_create_companies.rb:7]: as well as this one

File: app/models/company.rb [hunk 8]
```diff
@@ -0,0 +1,14 @@
+class Company < ApplicationRecord
+  enum company_type: { water_supply: 0, heating: 1, rent_payment: 2, accommodation_payment: 3,
```
Comment (chu4) [inline app/models/company.rb:2]: all symbols should be a constant
for later re-use

File: db/seeds.rb (hunk 4 above)
Comment (vasya539) [reply db/seeds.rb:10]: > add new line at end of file

most of the code editors do this for you automatically, just turn this feature on in settings

File: spec/requests/companies_request_spec.rb [hunk 9]
```diff
@@ -0,0 +1,10 @@
+require 'rails_helper'
+
+RSpec.describe "Companies", type: :request do
```
[the remark this answers is not in the document]
Comment (vasya539) [reply spec/requests/companies_request_spec.rb:3]: delete this file if you are not writing any requests specs, once again don't use generators
image

**Comment by DenisRem** [comment]
commit 2f23e81c2c01b7380315fa26a5a68a65b85fd13f

**Pull request closed without merging**